project_include.cmake

```diff
@@ -18,21 +19,34 @@ function(littlefs_create_partition_image partition base_dir)
 	partition_table_get_partition_info(size "--partition-name ${partition}" "size")
 	partition_table_get_partition_info(offset "--partition-name ${partition}" "offset")
 
-	add_custom_command(
-		OUTPUT ${MKLITTLEFS}
-		COMMAND make clean && make dist LFS_NAME_MAX=${CONFIG_LITTLEFS_OBJ_NAME_LEN}
-		WORKING_DIRECTORY ${MKLITTLEFS_DIR}
-		DEPENDS ${SDKCONFIG}
-	)
-
 	if("${size}" AND "${offset}")
 		set(image_file ${CMAKE_BINARY_DIR}/${partition}.bin)
 
+		if(CMAKE_HOST_WIN32)
+			set(littlefs_py "${littlefs_py_venv}/Scripts/littlefs-python.exe")
+			add_custom_command(
+					OUTPUT ${littlefs_py_venv}
+					COMMAND ${PYTHON} -m venv ${littlefs_py_venv} && ${littlefs_py_venv}/Scripts/pip.exe install -r ${littlefs_py_requirements}
+					WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+					DEPENDS ${littlefs_py_requirements}
+			)
+		else()
+			set(littlefs_py "${littlefs_py_venv}/bin/littlefs-python")
+			add_custom_command(
+					OUTPUT ${littlefs_py_venv}
+					COMMAND ${PYTHON} -m venv ${littlefs_py_venv} && ${littlefs_py_venv}/bin/pip install -r ${littlefs_py_requirements}
+					WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+					DEPENDS ${littlefs_py_requirements}
+			)
+		endif()
+
 		# Execute LittleFS image generation; this always executes as there is no way to specify for CMake to watch for
 		# contents of the base dir changing.
+
+		# TODO --name-max=${CONFIG_LITTLEFS_OBJ_NAME_LEN} --prog-size=${CONFIG_LITTLEFS_PAGE_SIZE}
```

Comment:
Can we add something like:
```
if(NOT CONFIG_LITTLEFS_OBJ_NAME_LEN EQUAL 255)
    message(FATAL_ERROR "CONFIG_LITTLEFS_OBJ_NAME_LEN must be 255 to precompile a filesystem image.")
endif()
```

And then I don't think there's anything TODO with the page size.